readthedocs/builds/forms.py

```diff
@@ -88,8 +89,22 @@ def _is_default_version(self):
         project = self.instance.project
         return project.default_version == self.instance.slug
 
+    def clean_slug(self):
+        slug = self.cleaned_data["slug"]
+        if "slug" in self.changed_data and self.instance.machine:
+            raise forms.ValidationError(
+                _("The slug cannot be changed."),
+            )
+        # TODO: check for unique slug instead of having django 500.
+        return slug
+
     def save(self, commit=True):
         obj = super().save(commit=commit)
```

Comment:
We should probably call `self.clean_resources()` before performing calling `.save()`. That way, we delete the old documentation and avoid serving old docs under the old URL.